src/levenshtein.rs

```diff
@@ -478,6 +504,52 @@ pub fn levenshtein_naive_k_with_opts(a: &[u8], b: &[u8], k: u32, trace_on: bool,
     Some((dp1[hi - lo - 1], Some(res)))
 }
 
+fn translate_str(chars: &mut Vec<char>, s: &str) -> Option<Vec<u8>> {
+    s.chars().map(|c| match chars.iter().position(|&d| c == d) {
+        Some(i) => Some(i as u8),
+        None => {
+            let idx = chars.len();
+            if idx < 256 {
+                chars.push(c);
+                Some(idx as u8)
+            } else {
+                None
+            }
+        }
+    }).collect()
+}
+
+fn all_ascii(s: &str) -> bool {
```

Comment:
Doesn't Rust already have an [`is_ascii`](https://doc.rust-lang.org/std/primitive.str.html#method.is_ascii) function?

Reply:
it does..